crates/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -8,14 +8,8 @@ use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum DefCollectorErrorKind {
-    #[error("duplicate function found in namespace")]
-    DuplicateFunction { first_def: Ident, second_def: Ident },
-    #[error("duplicate function found in namespace")]
-    DuplicateModuleDecl { first_def: Ident, second_def: Ident },
-    #[error("duplicate import")]
-    DuplicateImport { first_def: Ident, second_def: Ident },
-    #[error("duplicate global found in namespace")]
-    DuplicateGlobal { first_def: Ident, second_def: Ident },
+    #[error("duplicate {typ} found in namespace")]
+    Duplicate { typ: String, first_def: Ident, second_def: Ident },
```

Comment:
Nitpick: can we change `typ` to a `&'static str`?

Reply:
yes, I was wondering which is better, I will go with your suggestion and update the PR tomorrow :)

Reply:
Done